### ISSUE TITLE
Bad typing with Union query

### ISSUE THREAD

**Issue by MaximeDauphinot** [graphql-editor/graphql-zeus#218]
When there's the '__typename: true' in the request. the request completely loose the typing ( pic 1 ) but when it's commented ( pic 2 ) i get back the typing. 

running on the latest version 

picture 1 :

<img width="581" alt="Capture d’écran 2021-09-23 à 17 07 58" src="https://user-images.githubusercontent.com/24574934/134534186-3501f3b6-3bff-4475-868d-fec2bbfbf7c0.png">

picture 2 :

<img width="583" alt="Capture d’écran 2021-09-23 à 17 08 15" src="https://user-images.githubusercontent.com/24574934/134534222-b9859d11-f99f-4767-8be5-e857808761e2.png">

the request :
```ts
const { fileContent } = await this.client.query({
      fileContent: [
        { fileId },
        {
          __typename: true,
          "...on FileContentCloud": {
            parts: {
              contentUrl: true,
              order: true,
              md5: true
            },
            key: true,
            publicKey: true,
            totalSize: true,
            md5: true,
            md5Encrypted: true
          },
          "...on FileContentReceivedMail": {
            parts: {
              contentUrl: true,
              order: true,
              md5: true
            },
            key: true,
            senderPublicKey: true,
            totalSize: true,
            md5: true,
            md5Encrypted: true
          },
          "...on FileContentSentMail": {
            parts: {
              contentUrl: true,
              order: true,
              md5: true
            },
            key: true,
            totalSize: true,
            md5: true,
            md5Encrypted: true
          }
        }
      ]
    });
```

the api.graphql :

```
### This file was generated by Nexus Schema
### Do not make changes to this file directly


type AppKeyPair {
  pub: String!
}

type AppSession {
  city: String
  country: String
  createdAt: DateTime!
  id: ID!
  ip: String
  lastActivity: DateTime
  userAgent: String
  userApp: ApplicationsOnUsers
  userSession: UserSession!
}

type Application {
  id: ID!
  name: String!
}

type ApplicationsOnUsers {
  app: Application!
  fakeUserId: String!
  folders: [FoldersOnUsersOnApplications!]!
  keyPair: AppKeyPair
  sessions: [AppSession!]!
  settings: UserAppSettings!
  user: User!
}

"""
The `BigInt` scalar type represents non-fractional signed whole numeric values.
@see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt
"""
scalar BigInt

type Blog {
  entries: [BlogEntry!]!
  id: ID!
  order: Int!
}

type BlogEntry {
  author: String!
  body: String!
  date: DateTime!
  id: ID!
  image: String!
  lang: Lang!
  summary: String!
  title: String!
}

type BlogItem {
  author: String!
  blogId: String!
  body: String!
  date: DateTime!
  id: String!
  image: String!
  imageAlt: String!
  order: Int!
  summary: String!
  title: String!
}

"""The `Byte` scalar type represents byte value as a Buffer"""
scalar Bytes

type CloudLimits {
  count: BigInt!
  maxCount: BigInt
  maxSize: BigInt
  size: BigInt!
}

"""
A date-time string at UTC, such as 2007-12-03T10:15:30Z, compliant with the `date-time` format outlined in section 5.6 of the RFC 3339 profile of the ISO 8601 standard for representation of dates and times using the Gregorian calendar.
"""
scalar DateTime

"""An arbitrary-precision Decimal type"""
scalar Decimal

type Faq {
  entries: [FaqEntry!]!
  id: ID!
  order: Int!
}

type FaqEntry {
  answer: String!
  faq: Faq!
  id: ID!
  lang: Lang!
  question: String!
}

type FaqItem {
  answer: String!
  order: Int!
  question: String!
}

type File {
  createdAt: DateTime!
  deletedAt: DateTime
  id: ID!
  mails: [FilesOnMails!]!
  md5: String!
  md5Encrypted: String!
  size: BigInt!
  sizeBefore: BigInt!
  user: FilesOnUsersOnApplications
  userApp: ApplicationsOnUsers!
  users: [FilesOnUsersOnApplications!]!
  vFiles: [VFile!]!
  vFilesCurrent: [VFile!]!
  validatedAt: DateTime
}

union FileContent = FileContentCloud | FileContentReceivedMail | FileContentSentMail

interface FileContentBase {
  key: String!
  md5: String!
  md5Encrypted: String!
  parts: [FileContentPart!]!
  totalSize: BigInt!
  type: FileContentType!
}

type FileContentCloud implements FileContentBase {
  key: String!
  md5: String!
  md5Encrypted: String!
  parts: [FileContentPart!]!
  publicKey: String!
  totalSize: BigInt!
  type: FileContentType!
}

type FileContentPart {
  contentUrl: String!
  md5: String!
  order: Int!
}

type FileContentReceivedMail implements FileContentBase {
  key: String!
  md5: String!
  md5Encrypted: String!
  parts: [FileContentPart!]!
  senderPublicKey: String!
  totalSize: BigInt!
  type: FileContentType!
}

type FileContentSentMail implements FileContentBase {
  key: String!
  md5: String!
  md5Encrypted: String!
  parts: [FileContentPart!]!
  totalSize: BigInt!
  type: FileContentType!
}

enum FileContentType {
  cloud
  received_mail
  sent_mail
}

type FilePart {
  file: File!
  fileId: String!
  id: ID!
  md5: String!
  order: Int!
  size: BigInt!
  validatedAt: DateTime
}

type FilePartResponse {
  fields: Json!
  order: Int!
  url: String!
}

type FileResponse {
  fileId: String!
  filePartSize: BigInt!
  parts: [FilePartResponse!]!
}

type FileShared {
  appId: String!
  filename: String!
  id: String!
  message: String
  size: Int!
  sizeBefore: Int!
  userId: String!
}

type FileSharedContent {
  appId: String!
  contentUrl: String!
  filename: String!
  id: String!
  key: String!
  message: String
  ownerId: String
  pubKey: String
  size: Int!
  sizeBefore: Int!
  type: String!
}

type FilesOnMails {
  file: File!
  fileKey: String!
  filename: String!
  mail: Mail!
}

type FilesOnUsersOnApplications {
  file: File!

  """Encrypted with UserApp keys"""
  key: String!
  userApp: ApplicationsOnUsers!
}

type Folder {
  breadcrumb: [FolderBreadcrumbItem!]!
  createdAt: DateTime!
  createdBy: ApplicationsOnUsers!
  deletedAt: DateTime
  id: ID!
  isFavorite: Boolean!
  name: String!
  parentFolder: Folder
  subfolders: [Folder!]!
  updatedAt: DateTime!
  userApp: FoldersOnUsersOnApplications
  userApps: [FoldersOnUsersOnApplications!]!
  vFiles: [VFile!]!
}

type FolderBreadcrumbItem {
  id: String!
  name: String!
  nameKey: String
  pubKey: String!
}

type FolderSize {
  size: BigInt!
  sizeBefore: BigInt!
}

type FoldersOnUsersOnApplications {
  folder: Folder!
  isRoot: Boolean!
  nameKey: String
  rights: Rights!
  userApp: ApplicationsOnUsers!
}

"""
The `JSONObject` scalar type represents JSON objects as specified by [ECMA-404](http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf).
"""
scalar Json @specifiedBy(url: "http://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf")

enum Lang {
  de
  en
  es
  fr
  it
  ja
  nl
  pl
  pt
  ru
  zh
}

type Limits {
  downloadBandwidth: BigInt
  downloadCount: BigInt
  hardFileCount: BigInt
  hardFileSize: BigInt
  perFileSize: BigInt
  receivedMails: MailLimits!
  sentMails: MailLimits!
  uploadBandwidth: BigInt
  uploadCount: BigInt
}

type Mail {
  app: Application!
  body: String!
  createdAt: DateTime!
  deletedAt: DateTime
  files: [FilesOnMails!]!
  id: ID!
  mailIntegrity: MailIntegrity
  mailIntegrityDraft: MailIntegrityDraft
  openedAt: DateTime
  recipient: User!
  recipients: [User!]!
  sender: User!
  subject: String!
  type: MailType!
}

input MailFileInput {
  fileKey: String!
  id: String!
  name: String!
}

type MailIntegrity {
  answers: [MailIntegrity!]!
  app: Application!
  hash: String!
  hashKey: String!
  id: ID!
  mails: [Mail!]!
  recipients: [User!]!
  replyTo: MailIntegrity
  temporaryRecipients: [TemporaryUser!]!
}

type MailIntegrityDraft {
  app: Application!
  hash: String!
  hashKey: String!
  id: ID!
  mail: Mail!
  recipients: [User!]!
  replyTo: MailIntegrity
  temporaryRecipients: [TemporaryUser!]!
}

type MailLimits {
  count: BigInt
  fileCount: BigInt
  fileSize: BigInt
  hardCount: BigInt
  hardFileCount: BigInt
  hardFileSize: BigInt
  perFileSize: BigInt
}

type MailLimitsValues {
  count: BigInt!
  fileCount: BigInt!
  fileSize: BigInt!
  maxCount: BigInt
  maxFileCount: BigInt
  maxFileSize: BigInt
}

input MailRecipientInput {
  body: String!
  files: [MailFileInput!]!
  recipientId: String!
  subject: String!
}

enum MailType {
  received
  sent
}

type Mutation {
  addFileToHistory(fileId: String!, vFileId: String!): VFile!
  cancelPayment: Boolean
  createApplication(developerId: String!, name: String!, origin: String!): String
  createDraftMail(body: String!, hash: String!, hashKey: String!, recipients: [String!]!, replyTo: String, senderFiles: [MailFileInput!]!, subject: String!): Mail
  createFolder(key: String!, name: String!, parentFolderId: String): Folder
  dbConfig(appCode: String!, config: String!): Json!
  dbSet(userId: String, value: String!): Json
  deleteDraftMail(draftId: String!): Boolean
  deleteFile(fileId: String!, vFileId: String!): Boolean!
  deleteFolder(id: String!): Boolean
  deleteFolderCloudTrash(ids: [String!]!): Boolean
  deleteFolderSharing(folderId: String!, userId: String!): Boolean!
  deleteMail(mailId: String!): Boolean!
  deleteMailTrash(ids: [String!]!): Boolean!
  deleteUser(userId: String): Boolean
  deleteVFile(fileId: String!): Boolean!
  deleteVFileCloudTrash(ids: [String!]!): Boolean
  deleteVFileSharing(userId: String!, vFileId: String!): Boolean!
  duplicateVFile(customName: String, fileId: String!, folderId: String): Boolean!
  emptyCloudTrash: Boolean
  emptyMailTrash: Boolean
  logout(sessionId: String): Boolean
  moveFiles(filesId: [String!]!, parentFolderId: String): Boolean
  moveFolders(foldersId: [String!]!, parentFolderId: String): Boolean
  pay(autoRenew: Boolean, input: PayInput!, planCode: String!): Boolean!
  readMail(mailId: String!): Boolean
  recoverFile(fileId: String!): Boolean!
  recoverFolder(id: String!): Boolean
  recoverMail(mailId: String!): Boolean!
  saveInCloud(fileId: String!, filename: String!, folderId: String, key: String!, nameKey: String!): VFile!
  sendDraftMail(customMessage: String, draftMailId: String!, recipients: [MailRecipientInput!]!, temporaryRecipients: [String!]!): Boolean
  sendOneMail(mailIntegrityId: String!, recipient: MailRecipientInput!): Boolean
  sendReport(customMessage: String, reportedUserId: String!): Report
  shareFileInHistory(input: ShareFileInHistoryInput!, vFileId: String!): Boolean!
  shareFolder(folderId: String!, userId: String!): ShareFolderResponse!
  shareFolderFinish(rights: Rights!, shareFolders: ShareFoldersInput!, userId: String!): Boolean
  shareVFile(history: [ShareFileKeyInput!]!, nameKey: String!, rights: Rights!, userId: String!, vFileId: String!): Boolean
  unreadMail(mailId: String!): Boolean
  updateAppSettings(autoDisconnectDelay: Int, cloudFileDaysForDelete: Int, cloudFolderDaysForDelete: Int, historyFileDaysForDelete: Int, historyMaxFileCount: Int): UserAppSettings!
  updateDraftMail(body: String, draftId: String!, hash: String, hashKey: String, recipients: [String!], replyTo: String, senderFiles: [MailFileInput!], subject: String): Mail
  updateFile(deletedAt: DateTime, fileId: String!, filename: String, isFavorite: Boolean): VFile
  updateFolder(deletedAt: DateTime, folderId: String!, isFavorite: Boolean, name: String): Folder
  updateProfile(firstname: String, lang: Lang, lastname: String): User
  uploadAnonymous(fileName: String!, fileSize: Int!, fileSizeBefore: Int!, message: String, pubKey: String!, token: String!): FileResponse
  uploadFile(fileKey: String!, fileSize: BigInt!, fileSizeBefore: BigInt!, md5: String!, md5Encrypted: String!): FileResponse
  uploadFileEnd(fileId: String!): String
  uploadFilePartEnd(fileId: String!, md5: String!, order: Int!): Boolean!
}

input NameKeyInput {
  files: [ShareFileKeyInput!]!
  id: String!
  nameKey: String!
}

input NameKeyInputFolders {
  id: String!
  nameKey: String!
}

input PayInput {
  city: String!
  country: String!
  cvv: String!
  firstName: String!
  lastName: String!
  month: String!
  number: String!
  postalCode: String!
  street: String!
  token: String!
  type: PayInputType!
  year: String!
}

enum PayInputType {
  classic
  secure
}

type PaymentInfos {
  currentSubscription: RecurlySubscription!
  hostedUrl: String!
  invoices: [RecurlyInvoices!]!
}

type Plan {
  codes: [String!]!
  id: ID!
  kind: PlanKind!
  limits: Limits!
}

enum PlanKind {
  advanced
  basic
  free
  pro
}

type Query {
  blog(lang: Lang): [BlogItem!]!
  dbConfig(appCode: String!): Json
  dbGet(field: String!, userId: String): Json
  dbSearch(field: String!, search: String!): [Json!]!
  deletedFolders: [Folder!]!
  deletedMails(mailType: MailType!): [Mail!]!
  deletedVFiles: [VFile!]!
  faq(lang: Lang): [FaqItem!]!
  file(id: String!): File
  fileContent(fileId: String!): FileContent
  filesSharedWithMe(accepted: Boolean): [FileSharedContent!]!
  folder(deleted: Boolean, id: String): Folder
  folderSize(folderId: String): FolderSize
  foldersSharedWithMe: [Folder!]!
  limits: QueryLimits
  mail(id: String!): Mail
  paymentInfos: PaymentInfos
  plans: [Plan!]!
  sharedFolders: [Folder!]!
  sharedVFiles: [VFile!]!
  test: Boolean
  user(userId: String): User
  vFile(id: String!): VFile
  vFilesSharedWithMe: [VFile!]!
}

type QueryLimits {
  cloud: CloudLimits!
  mail: UserMailLimits!
}

type RecurlyInvoices {
  currency: String!
  date: DateTime!
  due: DateTime!
  name: String!
  pdf: String!
  status: String!
  total: Int!
}

type RecurlySubscription {
  activatedAt: DateTime
  autorenew: Boolean!
  canceledAt: DateTime
  collectionMethod: String!
  createdAt: DateTime
  currentPeriodEndsAt: DateTime
  currentPeriodStartedAt: DateTime
  expiresAt: DateTime
  pausedAt: DateTime
  plan: String
  planCode: String
  state: String!
  total: Int
  trialEndsAt: DateTime
  trialStartedAt: DateTime
  updatedAt: DateTime
}

type Report {
  customMessage: String
  id: ID!
  reportedUser: User!
  user: User!
}

enum Rights {
  admin
  read
  write
}

input ShareFileInHistoryInput {
  fileId: String!
  users: [ShareFileKeyInput!]!
}

input ShareFileKeyInput {
  id: String!
  key: String!
}

type ShareFolderResponse {
  folders: [String!]!
  vFiles: [String!]!
}

input ShareFoldersInput {
  folders: [NameKeyInputFolders!]!
  vFiles: [NameKeyInput!]!
}

type TemporaryUser {
  draftMails: [MailIntegrityDraft!]!
  email: String
  id: ID!
  mails: [MailIntegrity!]!
}

type User {
  appSettings: UserAppSettings
  applications: [ApplicationsOnUsers!]!
  createdAt: DateTime!
  deletedAt: DateTime
  draftMails: [Mail!]!
  email: String!
  firstname: String!
  godChildren: [User!]!
  godFather: User
  id: ID!
  lang: Lang!
  lastLogin: DateTime!
  lastname: String!
  phone: String!
  publicKey: String!
  receivedMails: [Mail!]!
  reportSent: [Report!]!
  role: UserRole!
  sentMails: [Mail!]!
  sessions: [UserSession!]!
  waitingReceivedMails: [WaitingReceivedMail!]!
}

type UserAppSettings {
  autoDisconnectDelay: Int
  cloudFileDaysForDelete: Int!
  cloudFolderDaysForDelete: Int!
  historyFileDaysForDelete: Int
  historyMaxFileCount: Int!
  id: ID!
}

type UserMailLimits {
  received: MailLimitsValues!
  sent: MailLimitsValues!
}

enum UserRole {
  admin
  user
}

type UserSession {
  appSessions: [AppSession!]!
  city: String
  country: String
  createdAt: DateTime!
  id: ID!
  ip: String
  lastActivity: DateTime
  userAgent: String
}

type VFile {
  createdAt: DateTime!
  createdBy: ApplicationsOnUsers!
  current: File
  deletedAt: DateTime
  filename: String!
  folder: Folder!
  history: [File!]!
  id: ID!
  isFavorite: Boolean!
  updatedAt: DateTime!
  userApp: VFilesOnUsersOnApplications
  userApps: [VFilesOnUsersOnApplications!]!
}

type VFilesOnUsersOnApplications {
  isRoot: Boolean!
  nameKey: String!
  rights: Rights!
  userApp: ApplicationsOnUsers!
  vFile: VFile!
}

type WaitingReceivedMail {
  attachmentsCount: Int!
  date: DateTime!
  recipients: [User!]!
  sender: User!
  temporaryRecipients: [TemporaryUser!]!
}
```

**Comment by aexol** [comment]
I will take a look today

**Comment by aexol** [comment]
found it fix in 30 mins

**Comment by aexol** [comment]
@MaximeDauphinot Solved in `3.2.3` Thanks for registering that ! 🧐

**Comment by MaximeDauphinot** [comment]
@aexol thanks you for this fix 🙂
